src/word-search/index.js

```diff
@@ -1,3 +1,115 @@
-function wordSearch(words, word) {}
+/*Steps to word search
+
+1. get index positions of each letter of word from words
+2. while visited not empty
+a. for each of the row,col pair, compute the adjacents left, adjacent bottom from the pair,
+b. updated visited with adjacent positon pair if is same as index position pair of word
+c. get alphabet representations of visited pairs from words
+d. if all word is in alpha reprensation- then the word was found
+*/
+
+// const words = [
+//   ["D", "S", "A", "N", "C"],
+//   ["I", "N", "O", "I", "T"],
+//   ["T", "A", "T", "R", "O"],
+//   ["M", "F", "O", "U", "T"],
+// ];
+
+// const word="SANCTIONS"
+
+function wordSearch(words, word) {
+  let wordIndex = [];
+  for (let p = 0; p < words.length; p++) {
+    for (m = 0; m < words[p].length; m++) {
+      if (word.includes(words[p][m])) {
+        wordIndex.push([p, m]);
+      }
+    }
+  }
+
+  
+  if (word[0] === word[word.length - 1]) {
+      //check if the word appears only once
+      let count = words.reduce(function (acc, curr) {
+        if (curr.includes(word[0])) {
+          acc += 1;
+        }
+        return acc;
+      }, 0);
+      if(count<2){
+        return false;
+      }
+  }
+
+  let i = 0;
+  let visited = [];
+  visited.push(wordIndex[0]); //initialize visited with first index array
+
+  while (i < visited.length) {
```

Comment:
put this while loop in a helper function and call that function, so that your code looks cleaner